CMakeLists.txt

```diff
@@ -241,35 +257,34 @@ add_definitions("-DQUERY_CANCELLATION_MODE=${QUERY_CANCELLATION_MODE}")
 set(USE_GNU_PARALLEL OFF CACHE BOOL "Don't use gnu parallel" FORCE)
 set(USE_OPENMP OFF CACHE BOOL "Don't use OPENMP as default" FORCE)
 
-add_subdirectory(third_party/stxxl EXCLUDE_FROM_ALL)
-
-# apply STXXL CXXFLAGS
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${STXXL_CXX_FLAGS}")
-include_directories(SYSTEM ${STXXL_INCLUDE_DIRS})
+set(STXXL_CXX_FLAGS "${STXXL_CXX_FLAGS} -Wno-deprecated-declarations")
+FetchContent_Declare(
+        stxxl
+        GIT_REPOSITORY https://github.com/ad-freiburg/stxxl.git
+        GIT_TAG 4d39edb272f10239c4302f89e448331cb8a57edb
+)
 
 ################################
 # RE2
 ################################
-# RE2 has a lot of unused-parameter warnings, we will deactivate
-# these for the subproject
-set(LOCAL_CXX_BACKUP_FLAGS "${CMAKE_CXX_FLAGS}")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
-
 set(RE2_BUILD_TESTING OFF CACHE BOOL "enable testing for RE2" FORCE)
-add_subdirectory(third_party/re2 EXCLUDE_FROM_ALL)
-target_compile_options(re2 PRIVATE -Wno-unused-but-set-variable)
-include_directories(SYSTEM third_party/re2)
-
+FetchContent_Declare(
+        re2
+        GIT_REPOSITORY https://github.com/google/re2.git
+        GIT_TAG bc0faab533e2b27b85b8ad312abf061e33ed6b5d # v.2023-11-01
+        GIT_SHALLOW TRUE
+        OVERRIDE_FIND_PACKAGE
+)
 
 ################################
 # Apply FetchContent
 ################################
-FetchContent_MakeAvailable(ctre)
+FetchContent_MakeAvailable( googletest ctre abseil re2 stxxl)
+# Disable some warnings in RE2 and STXXL
+target_compile_options(re2 PRIVATE -Wno-unused-parameter)
+target_compile_options(stxxl PRIVATE -Wno-deprecated-declarations)
 include_directories(${ctre_SOURCE_DIR}/single-header)
```

Comment:
Maybe this can be changed into a call of qlever_target_link_libraries with `ctre::ctre` if that works.